crates/sqlexec/src/planner/physical_plan/insert.rs

```diff
@@ -74,10 +71,14 @@ impl ExecutionPlan for InsertExec {
 
         let this = self.clone();
         let stream = stream::once(async move {
-            if this.table.meta.is_temp {
-                this.insert_temp(context).await
-            } else {
-                this.insert_native(context).await
+            match this.provider {
+                ProviderReference::RemoteReference(_) => Err(DataFusionError::Internal(
+                    "required table provider, found remote reference to insert".to_string(),
+                )),
+                ProviderReference::Provider(provider) => {
+                    // TODO: Add background job to track storage for native tables.
```

Comment:
is this in this PR (if so, maybe edit the description to include this?) otherwise, might this be it's own ticket in the epic?

Reply:
So we have background jobs, it got removed when working on RPC stuff, I'll create an issue to track these.